### PR TITLE
Added predict cli

### PR DESCRIPTION
**Pull request by chrlatte**
## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/samsledje/conplex-dti/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/samsledje/conplex-dti/blob/master/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: conplex_dti/cli/predict.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+from argparse import ArgumentParser
+
+import numpy as np
+import pandas as pd
+import torch
+from torch.utils.data import DataLoader
+
+from conplex_dti.featurizer import MorganFeaturizer, ProtBertFeaturizer
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: Would prefer to use relative imports here

`from ..featurizer import MorganFeaturizer...`
`from ..architectures import SimpleCoembedding...`

etc.

File: conplex_dti/cli/predict.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    )
+    parser.add_argument(
+        "--model-path",
+        type=str,
+        required=False,
+        default="./models/ConPLex_v1_BindingDB.pt",
+        help="Path to the file containing the model to use for predictions. Default: ./models/ConPLex_v1_BindingDB.pt",
+    )
+    parser.add_argument(
+        "--results-ofile",
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: Please update to just `--outfile`

File: conplex_dti/cli/predict.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        type=str,
+        required=False,
+        default="results.tsv",
+        help="Filename to save the resulting dataframe. Default: results.tsv",
+        # NOTE: the table is stored in the form [moleculeID] [proteinID] [prediction]
+    )
+    parser.add_argument(
+        "--features-dir",
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: `--data-dir`

File: conplex_dti/cli/predict.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    set_random_seed(args.random_seed)
+
+    try:
+        query_df = pd.read_csv(
+            args.data_file,
+            sep="\t",
+            names=["proteinID", "moleculeID", "proteinSequence", "moleculeSmiles"],
+        )
+    except FileNotFoundError:
+        print(f"Could not find data file: {args.data_file}")
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: Please replace print statements with `logg.info` -- see `cli/` train.py` for examples (including `from ..utils import config_logger`, etc.

File: conplex_dti/cli/predict.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    )
+    model.load_state_dict(torch.load(args.model_path, map_location=device))
+    model = model.eval()
+    model = model.to(device)
+
+    dt_feature_pairs = [
+        (drug_featurizer(r["moleculeSmiles"]), target_featurizer(r["proteinSequence"]))
+        for _, r in query_df.iterrows()
+    ]
+    dloader = DataLoader(dt_feature_pairs, batch_size=128, shuffle=False)
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: Can we add `batch_size` as a non-required parameter with default ~128 or 256? Should allow people to adapt to larger or smaller GPU memory.

**Review by samsledje — CHANGES_REQUESTED** [review]
This looks great Charlotte, thanks so much! I've added a few comments regarding stylistic and minor functional changes. Also, don't worry about it this time but in the future pull requests should be against the `dev` branch so that we can test their integration before pulling into `main`.  Thanks!

File: conplex_dti/cli/predict.py (hunk 3 above)
Comment (samsledje) [reply conplex_dti/cli/predict.py]: Sorry, I just checked with `train.py`, for consistency let's use `data-cache-dir` here

File: conplex_dti/cli/predict.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            args.data_file,
+            sep="\t",
+            names=["proteinID", "moleculeID", "proteinSequence", "moleculeSmiles"],
+        )
+    except FileNotFoundError:
+        print(f"Could not find data file: {args.data_file}")
+        return
+
+    print(f"Loading model from {args.model_path}")
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```
Comment (samsledje) [inline conplex_dti/cli/predict.py]: Device should also be a user-set argument

**Review by samsledje — CHANGES_REQUESTED** [review]

**Comment by chrlatte** [comment]
Hi Sam! Thanks for the feedback!

**Review by samsledje — APPROVED** [review]

**Pull request merged**